File: 1088.py

count = 0;
aux = 0;
while 1:
	vet = map(int, raw_input().split())
	if (vet[0] == 0): break

	for x in range(1,vet[0]):
		while(vet[x] != x):
			count += 2*(vet[x] - x) - 1
			aux = vet[vet[x]]
			vet[vet[x]] = vet[x]
			vet[x] = aux

	if (count % 2 == 0): 
		print "Carlos"
	else: 
		print "Marcelo"
	count = aux = n = 0;

File: 1088.cpp
#include <bits/stdc++.h>

using namespace std;


int main()
{
	int *vetor = new int[100];
	long int n;
	int aux, count = 0, x, j;
	while (1) {
		scanf("%ld", &n);
		if (n == 0) break;
		memset(vetor, 0, n);
		for (int i = 1; i <= n; ++i)
		{
			scanf("%d", &vetor[i]);
		}

		for (int i = 1; i <= n; ++i)
		{
			while (vetor[i] != i){
				count += 2*(vetor[i] - i) - 1;
				aux = vetor[vetor[i]];
				vetor[vetor[i]] = vetor[i];
				vetor[i] = aux;
			}
		}	


		if (count % 2 == 0) cout << "Carlos" << "\n";
		else cout << "Marcelo" << "\n";
		count = aux = n = 0;
	}
	return 0;
}
